daemon/algod/api/server/v2/handlers.go

```diff
@@ -181,17 +182,30 @@ func (v2 *Handlers) GetBlock(ctx echo.Context, round uint64, params generated.Ge
 		return ctx.Blob(http.StatusOK, "application/msgpack", blockbytes)
 	}
 
-	ledger := v2.Node.Ledger()
-	block, _, err := ledger.BlockCert(basics.Round(round))
-	if err != nil {
-		return internalError(ctx, err, errFailedLookingUpLedger, v2.Log)
-	}
+	if handle == protocol.JSONHandle {
```

Comment:
suggestion: instead of comparing the handle, add a switch statement above and switch by the params.Format instead. then, set the handle accordingly.